Added precombat option for stealth

glacialz opened the pull request:
[Rogue] Gave stealth the precombat option to take advantage of master of shadows.

glacialz committed:
Added precombat option for stealth

Gave stealth the precombat option to take advantage of master of shadows.

nuoHep commented on engine/class_modules/sc_rogue.cpp:
```diff
@@ -4102,9 +4102,15 @@ struct vendetta_t : public rogue_spell_t
 
 struct stealth_t : public rogue_spell_t
 {
+  double precombat_seconds;
```
Without evaluating if it's needed in the first place or not, times in simc are generally stored as `timespan_t`s.

glacialz replied:
It's cast to `timespan_t` after the conditional, similar to how precombat is implemented on the other spells in this file.

nuoHep replied:
Yes, I saw that. I'm not going to press it (not "my" module etc) but I'd argue the other spells are doing it "wrong" too.

navv1234 replied:
I'd agree with that sentiment. Time is intended to be represented by `timespan_t` to avoid ambiguity between units.

EvanMichaels commented on engine/class_modules/sc_rogue.cpp:
```diff
@@ -4117,6 +4123,14 @@ struct stealth_t : public rogue_spell_t
     trigger_master_of_shadows();
+
+    // trigger_master_of_shadows() doesn't trigger out of combat, must be applied manually
+    if ( precombat_seconds && !p()->in_combat )
```
I think either trigger_master_of_shadows() needs to be in an else case here, or this logic needs to be moved into the trigger_master_of_shadows() function itself with a precombat_seconds optional parameter or something. 

Think it's a little unclear to someone browsing this code as to why we are triggering MoS twice here or doing so manually when there is already a function here.

glacialz committed:
Updated trigger_master_of_shadows

Moved logic into trigger_master_of_shadows. Also made sure it checks for master of shadows talent

Mystler reviewed (commented):
I'm not really a fan of either the purpose nor the implementation. 1s prepull stealth is some pretty risky maneuver that I would call unrealistic, all for a super super minor energy minmax. As for the implementation, a copy of the trigger MoSh function is not really necessary and this could have been done with a default argument.
